### DISCUSSION TITLE
Connecting directly to the cluster peers

### DISCUSSION THREAD

**Discussion by Luflosi** [RubenKelevra/pacman.store#49] [Q&A]
`/ipns/cluster.pkg.pacman.store` contains five multiaddrs. I tried to see if I could connect to the three IPv4 ones directly using `ipfs swarm connect` but I always get errors like "Error: connect 12D3KooWDM4BGmkaxhLtEFbQJekdBHtWHo3ELUL4HE9f4DdNbGZx failure: context deadline exceeded".
I tried connecting from two different servers in two different locations.
Is this supposed to work?

**Comment by RubenKelevra** [comment]
Hey, sorry for the delay, the notifications for discussions were off.

We talked about this before, it was a remanence of an older iteration of this project. 

I fixed it in https://github.com/RubenKelevra/pacman.store/commit/09b33cc406dfc667f793b41a29e57097894b981b, so I think this can be "closed"? :)

**Reply by Luflosi** [reply]
It still doesn't work as I expect it to. Executing `ipfs swarm connect` to any of the IPv4 peer addresses listed in `service.json` doesn't work.
```
$ ipfs swarm connect /ip4/45.83.104.156/tcp/16587/p2p/12D3KooWDM4BGmkaxhLtEFbQJekdBHtWHo3ELUL4HE9f4DdNbGZx
$ ipfs swarm connect /ip4/94.176.233.122/tcp/16587/p2p/12D3KooWGCSifNrJPZPfEdhAjRtxCW2dukiqQEqg4RAV6hE2jgbA
```
prints one of the following (output for the second command omitted as it is basically the same):
```
Error: connect 12D3KooWDM4BGmkaxhLtEFbQJekdBHtWHo3ELUL4HE9f4DdNbGZx failure: failed to dial 12D3KooWDM4BGmkaxhLtEFbQJekdBHtWHo3ELUL4HE9f4DdNbGZx:
  * [/ip4/45.83.104.156/tcp/16587] dial tcp4 0.0.0.0:4001->45.83.104.156:16587: i/o timeout
```
```
Error: connect 12D3KooWDM4BGmkaxhLtEFbQJekdBHtWHo3ELUL4HE9f4DdNbGZx failure: context deadline exceeded
```
```
Error: connect 12D3KooWDM4BGmkaxhLtEFbQJekdBHtWHo3ELUL4HE9f4DdNbGZx failure: failed to dial 12D3KooWDM4BGmkaxhLtEFbQJekdBHtWHo3ELUL4HE9f4DdNbGZx:
  * [/ip4/45.83.104.156/tcp/16587] failed to negotiate security protocol: message did not have trailing newline
```
and
```
$ ipfs swarm connect /ip4/140.238.249.171/tcp/16587/p2p/12D3KooWN9pSnzmE57cN4XDoYPRs1htjSRF2ig528kwb3VRrvf1Z
```
prints
```
Error: connect 12D3KooWN9pSnzmE57cN4XDoYPRs1htjSRF2ig528kwb3VRrvf1Z failure: failed to dial 12D3KooWN9pSnzmE57cN4XDoYPRs1htjSRF2ig528kwb3VRrvf1Z:
  * [/ip4/140.238.249.171/tcp/16587] dial tcp4 140.238.249.171:16587: connect: no route to host
```
and
```
$ ipfs swarm connect /p2p/12D3KooWL3V95pdTwumVCqbxWzuyCud3DqawJBi7koGos1iLwESQ
```
prints
```
Error: connect 12D3KooWL3V95pdTwumVCqbxWzuyCud3DqawJBi7koGos1iLwESQ failure: routing: not found
```
whereas connecting to my own peer from a second peer works:
```
$ ipfs swarm connect /dns4/luflosi.de/udp/4001/quic/p2p/12D3KooWBqQrnTqx9Wp89p2bD1hrwmXYJQ5x1fDfigRCfZJGKQfr
```
```
connect 12D3KooWBqQrnTqx9Wp89p2bD1hrwmXYJQ5x1fDfigRCfZJGKQfr success
```
I executed every command except the last one on two machines in two different networks.
I also find it a bit odd, that the list of trusted peers contains peer `12D3KooWL3V95pdTwumVCqbxWzuyCud3DqawJBi7koGos1iLwESQ`, which is not listed anywhere else. Peer `12D3KooWN9pSnzmE57cN4XDoYPRs1htjSRF2ig528kwb3VRrvf1Z` is listed in `service.json` but not in the README. The peer run by @teknomunk starts with `12D3KooWQkhWQYp` according to the git history of the README but is not listed in `service.json`.

**Reply by RubenKelevra** [reply]
Well, that's expected. IPFS cannot connect to the IPFS-Cluster ports. :)

**Reply by RubenKelevra** [reply]
@Luflosi is your question now answered? :)

**Comment by RubenKelevra** [comment]
Closed and locked due to inactivity.

You can always connect to the servers in the list published in the README via:

```console 
$ ipfs swarm connect /ipv4/$ip/443/$ipfs_id
```

or

```console
$ ipfs swarm connect /ipv6/$ip/443/$ipfs_id
```



The port you used before is the port for the cluster daemon connections, which isn't going to work with IPFS itself. Same goes for the IDs. This are cluster member IDs.

Apart from that, this shouldn't really be necessary, as the cluster members are all running gossipsub as pubsub router. So they should resolve extremely quickly to a small list of nodes which take part in the IPNS.
